scripts/setup_osx.sh

```diff
@@ -1,38 +1,248 @@
+  boost-python \
+  ffmpeg \
+  sdl2 \
+  swig \
+  openmpi
+
+# Build GLFW because the brew version is too old
+GLFW_DIR="$(mktemp -d)/glfw"
+git clone https://github.com/glfw/glfw.git "${GLFW_DIR}"
+cd "${GLFW_DIR}"
+git checkout 0be4f3f75aebd9d24583ee86590a38e741db0904
+mkdir glfw-build
+cd glfw-build
+cmake -DBUILD_SHARED_LIBS=ON -DGLFW_BUILD_EXAMPLES=OFF \
+  -DGLFW_BUILD_TESTS=OFF -DGLFW_BUILD_DOCS=OFF ..
+make -j"$(nproc)"
+make install
+cd "${GARAGE_DIR}"
+
+# Leave a note in ~/.bashrc for the added environment variables
+if [[ "${_arg_set_envvar}" = on ]]; then
+  echo -e "\n# Added by the garage installer" >> "${BASH_PROF}"
 fi
 
-echo "Creating conda environment..."
+# Set up MuJoCo
+if [[ ! -d "${HOME}"/.mujoco/mjpro150 ]]; then
+  mkdir "${HOME}"/.mujoco
+  MUJOCO_ZIP="$(mktemp --suffix=_mujoco.zip)"
+  wget https://www.roboti.us/download/mjpro150_osx.zip -O "${MUJOCO_ZIP}"
+  unzip -u "${MUJOCO_ZIP}" -d "${HOME}"/.mujoco
+else
+  print_warning "MuJoCo is already installed"
+fi
+# Configure MuJoCo as a shared library
+export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:${HOME}/.mujoco/mjpro150/bin"
+LD_LIB_ENV_VAR="LD_LIBRARY_PATH=\"\$LD_LIBRARY_PATH:${HOME}/.mujoco/mjpro150"
+LD_LIB_ENV_VAR="${LD_LIB_ENV_VAR}/bin\""
+if [[ "${_arg_set_envvar}" = on ]]; then
+  echo "export ${LD_LIB_ENV_VAR}" >> "${BASH_PROF}"
+fi
+
+# Set up conda
+hash conda 2>/dev/null || {
+  CONDA_INSTALLER="$(mktemp --suffix=_miniconda.sh)"
+  sudo chmod u+x "${CONDA_INSTALLER}"
+  wget https://repo.continuum.io/miniconda/Miniconda2-latest-MacOSX-x86_64.sh \
+    -O "${CONDA_INSTALLER}"
+  sudo bash "${CONDA_INSTALLER}" -b
+  # Add conda to executable programs
+  CONDA_PATH="${HOME}/miniconda2/bin/"
+  export PATH="$PATH:${CONDA_PATH}"
+  PATH_ENV_VAR="PATH=\"\$PATH:${CONDA_PATH}\""
+  if [[ "${_arg_set_envvar}" = on ]]; then
+    echo "export ${PATH_ENV_VAR}" >> "${BASH_PROF}"
+  fi
+}
+conda update -q -y conda
+
+# Create conda environment
 conda env create -f environment.yml
-conda env update
+if [[ "${?}" -ne 0 ]]; then
+  print_error "Error: conda environment could not be created"
+fi
+
+# Extras
+source activate garage
+# Prevent pip from complaining about available upgrades
+pip install --upgrade pip
+# Fix Box2D install
+# See https://github.com/openai/gym/issues/100
+# See https://github.com/pybox2d/pybox2d/issues/82
+pip uninstall -y Box2D Box2D-kengz box2d-py
+pip install Box2D
+# We need a MuJoCo key to import mujoco_py
+cp ${_arg_mjkey_path} "${HOME}"/.mujoco/mjkey.txt
+python -c 'import mujoco_py'
+# Set up pre-commit in local repo
+pre-commit install -t pre-commit
+pre-commit install -t pre-push
+pre-commit install -t commit-msg
+source deactivate
+
+# Add garage to python modules
+export PYTHONPATH="$PYTHONPATH:${GARAGE_DIR}"
+PYTHON_ENV_VAR="PYTHONPATH=\"\$PYTHONPATH:${GARAGE_DIR}\""
+if [[ "${_arg_set_envvar}" = on ]]; then
+  echo "export ${PYTHON_ENV_VAR}" >> "${BASH_RC}"
+else
+  echo "Remember to export the following environment variables before" \
```

Comment:
Can you add an extra newline before this line? Or else it looks like it's part of the pre-commit install:

```
pre-commit installed at /private/var/folders/kj/dgb2_y1n1wj515xphc6clvzm0000gn/T/tmp.DWwDXgPH/glfw/.git/hooks/commit-msg
Remember to export the following environment variables before running garage:
```

Reply:
It looks like you are installing pre-commit in the temporary git checkout for building glfw? THat seems not useful.

Also, you should (hopefully) not need to compile glfw for Mac. Frankly I'm surprised that worked.

Reply:
I will.

Reply:
The pre-commit happened there because readlink failed as pointed in a comment above, so cd does not change to the garage dir.
I thought we were compiling glfw to include the newest changes since their last release is too old. Is glfw already included in Mac or installing from brew is good enough?